### PR TITLE
[dynamo] Remove `closure_cells` and merge/remove code paths

### PR DESCRIPTION
**Pull request by StrongerXi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #140155
* __->__ #140154
* #140153
* #140435
* #140436
* #140152
* #140330

Now that all cells are modeled as `NewCellVariable` in Dynamo, we no
longer need to put cell variables into this special `closure_cells`,
rather we just merge `closure_cells` with `symbolic_locals`.

This allows us to merge and remove some code paths, notably make
`LOAD_CLOSURE` the same as `LOAD_FAST`, and `LOAD_DEREF` & `STORE_DEREF`
the same for inlining or regular `InstructionTranslator`.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Comment by StrongerXi** [comment]
Rebase.

**Review by jansel — APPROVED** [review]

File: torch/_dynamo/symbolic_convert.py [hunk 1]
```diff
@@ -1069,40 +1069,25 @@ def pop(self) -> VariableTracker:
...
             self.push(self.symbolic_locals[name].unwrap(), name=name)
         except KeyError:
-            if sys.version_info >= (3, 13) and name in self.cell_and_freevars():
-                # 3.13 merged LOAD_CLOSURE into LOAD_FAST
-                # If we fail to LOAD_FAST, then we probably should have done LOAD_CLOSURE.
-                # Closure variable creation is actually done in SET_FUNCTION_ATTRIBUTE,
-                # but we'll do it again here so that we don't need to push a dummy variable.
-                # We shouldn't actually be doing anything with this variable anyway.
-                self.push(self._load_closure(name), name=name)
-            elif name.startswith("."):
```
Comment (williamwen42) [inline torch/_dynamo/symbolic_convert.py:1089]: Do we still need this condition?

Comment (StrongerXi) [reply torch/_dynamo/symbolic_convert.py:1089]: Honestly idk, but happy to take a look. This doesn't seem immediately related to closure cells so I didn't touch it.

Comment (williamwen42) [reply torch/_dynamo/symbolic_convert.py:1089]: guess if the tests are passing, it's probably fine then?

Comment (StrongerXi) [reply torch/_dynamo/symbolic_convert.py:1089]: Oh I didn't realize I removed the condition lol. Lemme bring it back just to be safe, I'll take a look into this renaming separately.

**Pull request closed without merging**